gptel-anthropic.el

```diff
@@ -283,29 +283,30 @@ TOOL-USE is a list of plists containing tool names, arguments and call results."
           ;; We check for blank prompts by skipping whitespace and comparing
           ;; point against the previous.
           (unless (save-excursion (skip-syntax-forward " ") (>= (point) prev-pt))
+            ;; XXX update for tools
             (pcase (get-char-property (point) 'gptel)
               ('response
-               (push (list :role "assistant"
-                           :content (buffer-substring-no-properties (point) prev-pt))
-                     prompts))
+               (when-let* ((content
+                            (gptel--trim-prefixes
+                             (buffer-substring-no-properties (point) prev-pt))))
+                 (when (not (string-blank-p content))
+                   (push (list :role "assistant" :content content) prompts))))
               ('nil                     ; user role: possibly with media
-               (if include-media       
-                   (push (list :role "user"
-                               :content
-                               (gptel--anthropic-parse-multipart
-                                (gptel--parse-media-links major-mode (point) prev-pt)))
-                         prompts)
-                 (push (list :role "user"
-                             :content
-                             (gptel--trim-prefixes
-                              (buffer-substring-no-properties (point) prev-pt)))
-                       prompts)))))
+               (if include-media
+                   (when-let* ((content (gptel--anthropic-parse-multipart
+                                         (gptel--parse-media-links major-mode (point) prev-pt))))
+                     (when (> (length content) 0)
+                       (push (list :role "user" :content content) prompts)))
+                 (when-let* ((content (gptel--trim-prefixes
+                                       (buffer-substring-no-properties (point) prev-pt))))
+                   (push (list :role "user" :content content) prompts))))))
           (setq prev-pt (point))
           (and max-entries (cl-decf max-entries)))
-      (push (list :role "user"
-                  :content
-                  (string-trim (buffer-substring-no-properties (point-min) (point-max))))
-            prompts))
+      (when-let* ((content (string-trim (buffer-substring-no-properties
+                                         (point-min) (point-max)))))
+        ;; XXX Remove this comment if empty `prompts' is fine.
+        (when (not (string-empty-p content))
```

Comment:
If the content is empty here and it's included in messages, the request is going to fail.  If content is empty and it's not included in messages, the request is going to fail.  So it's just a question of choosing which failure we prefer.  I prefer sending an empty string to not sending a user prompt at all -- the latter is more confusing to the user.  So please remove this `string-empty-p` check.

Reply:
Hmmm  If both cases are definitely going to fail, shouldn't we just `error` immediately since we can explain it directly?  Should we do that on every backend?  Thanks for checking that XXX.

Reply:
Possibly. I haven't considered how to handle throwing an error in `gptel--parse-buffer`.  Errors in the response are handled more gracefully.

In any case please don't add it to this PR.  There are already too many changes here.